web/packages/shared/components/Notification/Notification.tsx

```diff
@@ -113,79 +116,137 @@ export function Notification(props: NotificationProps) {
-          {content.list && <List items={content.list} />}
-          {content.description}
-          {content.link && (
-            <Link
-              css={`
-                display: block;
-              `}
-              href={content.link.href}
-              target="_blank"
-              onClick={e => e.stopPropagation()} // prevents notification from collapsing
-            >
-              {content.link.text}
-            </Link>
-          )}
-        </Text>
-      </Flex>
-    );
+  switch (severity) {
+    case 'success':
+      return <Icon.Checks aria-label="Success" {...commonProps} />;
+    case 'error':
+      return <Icon.WarningCircle aria-label="Danger" {...commonProps} />;
+    case 'info':
+      return <Icon.Info aria-label="Info" {...commonProps} />;
+    case 'warn':
+      return <Icon.Warning aria-label="Warning" {...commonProps} />;
+    case 'neutral':
+      return <Icon.Notification aria-label="Note" {...commonProps} />;
+    default:
+      severity satisfies never;
   }
-}
+};
+
+const toObjectContent = (
+  content: NotificationItemContent
+): NotificationItemObjectContent =>
+  typeof content === 'string' ? { description: content } : content;
+
+const notificationColors = (theme: Theme, severity: NotificationSeverity) => {
+  switch (severity) {
+    case 'neutral':
+      return {
+        borderColor: theme.colors.interactive.tonal.neutral[2],
+        iconColor: theme.colors.text.main,
+      };
+    case 'error':
+      return {
+        borderColor: theme.colors.interactive.solid.danger.default.background,
+        iconColor: theme.colors.interactive.solid.danger.default.background,
+      };
+    case 'warn':
+      return {
+        borderColor: theme.colors.interactive.solid.alert.default.background,
+        iconColor: theme.colors.interactive.solid.alert.default.background,
+      };
+    case 'info':
+      return {
+        borderColor: theme.colors.interactive.solid.accent.default.background,
+        iconColor: theme.colors.interactive.solid.accent.default.background,
+      };
+    case 'success':
+      return {
+        borderColor: theme.colors.interactive.solid.success.default.background,
+        iconColor: theme.colors.interactive.solid.success.default.background,
+      };
+    default:
+      severity satisfies never;
+  }
+};
+
+const CloseIcon = styled(ButtonIcon)`
+  /* Place the close button so that its hover state circle "overflows" the layout
+   * padding a bit, placing the close icon itself in alignment with the padding. */
+  position: absolute;
+  top: ${props => props.theme.space[2]}px;
+  right: ${props => props.theme.space[2]}px;
+`;
+
+const NotificationBody = ({
+  content,
+  isExpanded,
+}: {
+  content: NotificationItemObjectContent;
+  isExpanded: boolean;
+}) => {
+  const longerTextCss = isExpanded ? textCss : shortTextCss;
+
+  return (
+    <>
+      <Text typography="body2" color="text.slightlyMuted" css={longerTextCss}>
```

Comment:
Should notifications with only the title be allowed? If so, then I think we should render this `<Text>` only if either `content.list` or `content.description` is specified. Otherwise it adds a lot of unnecessary space which makes the notification look off:

<img width="342" alt="only-title" src="https://github.com/user-attachments/assets/26fe5713-c63e-4ae9-969d-7d60e846d2b4">

Speaking of, we have a bunch of notifications in Connect which use just the description, like the one about the config file in your screenshot or this one about tsh in path (this is in the three dot menu in the top right as well):

<img width="1044" alt="no-title" src="https://github.com/user-attachments/assets/b0343362-dff3-4e53-87c6-8fd846804c6e">

I personally don't like it this much. This layout looks good in Figma where the close button is always shown, the description has multiple lines and there's an action button. But in simple cases like this one, I feel like it wastes a lot of space. Should we make it so that `toObjectContent` puts the string into title and not description?

<img width="1044" alt="just-title" src="https://github.com/user-attachments/assets/5001c09a-0173-4a4d-a93d-e26763745a21">

(In this screenshot, I fixed the issue with spacing that this `<Text>` presents.)

Reply:
Oops, I didn't notice that extra space in the bottom. I'll fix it. As for whether we should use title-only or description-only notifications: there's a tradeoff here, since though single-line messages look better as titles, multiline ones look better with the intended design. @roraback or @rishibarbhaya-design , do you have any opnion here? If not, tomorrow I'll change the default to what @ravicious proposed. (The description-only UI will still be available, just not picked by default, but I still think it's better to have a default convention.)